reolink_aio/api.py

```diff
@@ -3614,7 +3614,8 @@ async def send_chunk(
                 data = await response.read()  # returns bytes
             elif expected_response_type == "application/octet-stream":
                 async with self._send_mutex:
-                    response = await self._aiohttp_session.get(url=self._url, params=param, allow_redirects=False)
+                    # disable timeout because downloads can be large
+                    response = await self._aiohttp_session.get(url=self._url, params=param, allow_redirects=False, timeout=None)
```

Comment:
No timeout at all will mean the code can possibly hang here for ever and get stuck.
For instance when the network goes down during download or something, it will wait forever on a connection.

I would suggest a timeout of 5 minutes or something larger if needed, maybe even 15 or 30 minutes, but do keep some timeout.

When downloading a 1 hour 4K stream, how long does that take on your machine?

Reply:
Unfortunately, there is no practical timeout to supply as it is a total timeout, not a no data timeout, so however long you set it to, someone will have a transfer that will take longer. However, on the alternative, it is possible to use  localized timeout in the read operation (on the consumer end of the stream.) Since that would run in chunks, it would be possible to put chunk reads in a timeout, but that is outside of this call. I do know, over wifi, the downloads do go in bursts, though the largest I tested with was a 10 min capture and it took about 3-4min to download. 

I feel that documenting that there is no default timeout on the stream and it is up to the consumer to provide proper handling.
This is more of a special use case anyway, so it is not something a general user should ever be concerned about.

Reply:
See https://docs.aiohttp.org/en/stable/client_reference.html?highlight=timeout#clienttimeout:
you can more finely control the timeout:
```suggestion
                    timeout = aiohttp.ClientTimeout(total=None, connect=self.timeout, sock_read=self.timeout)
                    response = await self._aiohttp_session.get(url=self._url, params=param, allow_redirects=False, timeout=timeout)
```

This should work and provide better safety for not getting stuck.
Still I would prefer to have the total timeout set to something like 1 hour (3600s) to at least make sure it can't get stuck forever. But If you think that might still cause issues we can leave total=None.

Reply:
I tested with a 30/30 (conn/read) and it did not throw a problem. With the two set, the only way you could miss a timeout is if the consumer just sat with the stream and did nothing. 

I want to avoid a total timeout, because you do not know what could cause a delay, If someone was downloading an hour long video (unlikley but you never know in the future) and the downloader pauses for a minute for playback to catch up, this could cause it to fail. Thats why I said an arbitrary overall timeout is not a good idea. It only works when you are absolutely sure no possible operation could take longet than the timeout. At an API/library level there is no way to gaurentee this, you have to rely on the consumer to be responsible.